ft_virtualchannel.m

```diff
@@ -0,0 +1,382 @@
+        tmpdata{i}.(sensfields{k}) = ft_apply_montage(tmpdata{i}.grad, montage, 'feedback', 'none', 'keepunused', 'no', 'balancename', bname);
+      end
+    end
+    
+    compmethods = {'pca' 'runica' 'fastica' 'dss'};
+    switch cfg.method
+      case compmethods
+        
+        tmpcfg     = keepfields(cfg, {'method' cfg.method 'numcomponent'});
+        if isequal(tmpcfg.numcomponent, 'all')
+          tmpcfg = rmfield(tmpcfg, 'numcomponent');
+        end
+        
+        tmpdata{i}  = ft_componentanalysis(tmpcfg, tmpdata{i});
+        unmixing{i} = tmpdata{i}.unmixing * unmixing{i};
+        
+        for k = 1:numel(tmpdata{i}.label)
+          if usepos
+            str = sprintf('virtualchannel%03d', i);
+          else
+            str = cfg.channel{i};
+          end
+        end
+        tmpdata{i}.label{k} = sprintf('%s_%s%03d', str, cfg.method, k);
+      
+      case 'none'
+        % do nothing
+      
+    otherwise
+      ft_error('currently not yet supported');
+      % the idea would to support a custom function here, with a
+      % function(cfg.(cfg.method), tmpdata{i}) API
+      
+    end % reduction of components
+  
+  else
+    
+    [i1, i2] = match_str(data.label, montage.labelold);
+    
+    % ensure that the used channels run from 1:numel(data.label), to be
+    % sure that the to-be-computed covariance matches the labels, not sure
+    % whether this is needed, this could probably also be achieved by doing
+    % a check on the input data's labels, related to the source.filter
+    % labels
+    assert(isequal(i1(:)', 1:numel(data.label)));
+    
+    unmixing{1,i}(:,i1) = montage.tra(:,i2);
+    topolabel{1,i}      = data.label(i1);
+    
+    if ~exist('tlck', 'var')
+      % create a data structure that contains a covariance -> FIXME requires
+      % a check on the input data, for now assume raw data
+      tmpcfg = [];
+      tmpcfg.covariance = 'yes';
+      tlck = ft_timelockanalysis(tmpcfg, data);
+    end  
+    
+    % create a matricial square root of the full covariance matrix
+    [u, s, v] = svd(tlck.cov);
+    Csqrtm    = u*sqrt(s);
+     
+    % don't do the sandwiching for efficiency, we will use only the u
+    % matrix, this yields the same u matrix as the svd on the tlck.cov
+    [u, s, v] = svd(unmixing{1, i} * Csqrtm, 'econ');
+    
+    if isequal(cfg.numcomponent, 'all')
+      ncomp = size(u,2);
+    else
+      if size(u,2)>=cfg.numcomponent
+        ncomp = cfg.numcomponent;
+      else
+        ncomp = size(u,2);
+        ft_warning(sprintf('using %d components,rather than the requested %d\n', ncomp, cfg.numcomponent));
+      end
+    end
+    
+    unmixing{1, i} = u(:,1:ncomp)' * unmixing{1, i};
+    label{1, i}    = cell(ncomp,1);
+    if usepos
+      str = sprintf('virtualchannel%03d', i);
+    else
+      str = cfg.parcel{i};
+    end
+    for k = 1:ncomp
+      label{1, i}{k} = sprintf('%s_svd%03d', str, k);
+    end
```

Comment:
Same as above, special case for `numcomponent == 1`.

```suggestion
    if ncomp == 1
      label{1, i}{1} = str;
    else
      for k = 1:ncomp
        label{1, i}{k} = sprintf('%s_svd%03d', str, k);
      end
    end
```

Reply:
same comment to this as above :)